### PR TITLE
populate tzip-16 contract data when contracts are originated

### PR DESCRIPTION
**Pull request by emmanueldenloye**
The spec only requires that the 'description' and 'interfaces' fields
to be defined IIRC

This should resolve issue #131

### CODE REVIEW

File: contracts/ligo/src/tzip16-compat/fa2_multi_nft_asset.mligo [hunk 1]
```diff
@@ -48,7 +48,11 @@ let sample_storage : nft_asset_storage = {
     pending_admin = (None : address option);
     paused = true;
     };
-  metadata = (Big_map.empty : (string, bytes) big_map);
+  metadata = Big_map.literal
+               [
+                 ("description", Bytes.pack "sample_token");
+                 ("interfaces", Bytes.pack ["TZIP-016"])
```
Comment (emishur) [inline contracts/ligo/src/tzip16-compat/fa2_multi_nft_asset.mligo:54]: It also implements TZIP-012. This is what would be used by indexers and wallets to know that this is a token contract

File: contracts/ligo/src/tzip16-compat/fa2_multi_nft_faucet.mligo [hunk 2]
```diff
@@ -26,5 +26,9 @@ let sample_storage : nft_faucet_storage = {
     next_token_id = 0n;
     operators = (Big_map.empty : operator_storage);
     };
-  metadata = (Big_map.empty: (string, bytes) big_map);
+  metadata = Big_map.literal
+               [
+                 ("description", Bytes.pack "sample_faucet_token");
+                 ("interfaces", Bytes.pack ["TZIP-016"])
```
Comment (emishur) [inline contracts/ligo/src/tzip16-compat/fa2_multi_nft_faucet.mligo:32]: also implements TZIP-012

File: contracts/src/nft-contracts-tzip16.ts [hunk 3]
```diff
@@ -37,8 +37,10 @@ export async function originateNft(
         'nft_asset_main',
         'fa2_multi_nft_asset_tzip16_compat.tz'
     );
-    const storage = `(Pair (Pair (Pair "${admin}" False) None)
-  (Pair (Pair {} 0) (Pair {} {})))`;
+    const storage = `(Pair (Pair (Pair (Pair "${admin}" True) None)
+            (Pair (Pair {} 0) (Pair {} {})))
+      { Elt "description" 0x05010000000c73616d706c655f746f6b656e ;
```
Comment (emishur) [inline contracts/src/nft-contracts-tzip16.ts:42]: Does not look like correct TZIP-16 metadata map format.
It should have and entry with the key "" pointing to metadata uri (for the metadata in the same big_map the value should be byte-encoded "tezos-storage:content") and a second key ("content") which pints to byte-encoded JSON containing all metadata fields inside.
Consider specifying plain metadata JSON first and then encode it. This way it is possible to verify the content from reading source code

**Review by emishur — CHANGES_REQUESTED** [review]

File: contracts/ligo/src/tzip16-compat/fa2_multi_nft_faucet.mligo [hunk 4]
```diff
@@ -20,15 +20,19 @@ let nft_faucet_main (param, storage : nft_faucet_entrypoints * nft_faucet_storag
      let ops, new_assets = mint_tokens (mp, storage.assets) in
      ops, { storage with assets = new_assets; }
 
-let sample_storage : nft_faucet_storage = {
-  assets = { ledger = (Big_map.empty : ledger);
+let sample_storage : nft_faucet_storage =
+  let contents_name = "" in
+  let contents_description = "" in
+  let contents_interfaces = "" in
+  let contents = Bytes.pack("{" ^ contents_name ^ "," ^ contents_description ^ "," ^ contents_interfaces ^ "}") in
```
Comment (emishur) [inline contracts/ligo/src/tzip16-compat/fa2_multi_nft_faucet.mligo:27]: content should be well formed JSON.
See https://github.com/tqtezos/smart-contracts/blob/9796871b8a5ba7c85587397afff6a9ccf483005c/fractional/test/origination.ts#L23-L31 and https://github.com/tqtezos/smart-contracts/blob/9796871b8a5ba7c85587397afff6a9ccf483005c/fractional/test/origination.ts#L73 for example

Comment (emmanueldenloye) [reply contracts/ligo/src/tzip16-compat/fa2_multi_nft_faucet.mligo:27]: This was an error. I thought I had made the same change here as I did in another file. I'll update this shortly.

**Review by emishur — APPROVED** [review]

**Pull request merged**